consensus/src/transaction_shuffler/sender_aware/mod.rs

```diff
@@ -29,200 +81,48 @@ use std::collections::{HashMap, VecDeque};
 ///         else we add it to the block
 ///   else
 ///       take the first transaction from the pending transactions and add it to the block
-
-pub struct SenderAwareShuffler {
-    conflict_window_size: usize,
-}
-
 impl TransactionShuffler for SenderAwareShuffler {
     fn shuffle(&self, txns: Vec<SignedTransaction>) -> Vec<SignedTransaction> {
         // Early return for performance reason if there are no transactions to shuffle
+        // TODO make sure this is handled in iterator
```

Comment:
Do these TODOs need test cases?

Reply:
Added tests and changed some of the logic to handle the edge cases 

1. `conflict_window_size == 0` and 
2. `txns.len() == 0`